### ISSUE TITLE
Issues with fields from pass.json->"storeCard" not being added to compiled pass

### ISSUE THREAD

**Issue by djpimp2010** [alexandercerutti/passkit-generator#203]
## Running OS

MacOS node (this will eventually be deployed inside a linux docker)

## Running Node Version

18.19.1

## Description

I am signing a pass with the type of "storeCard" and the signed stream is returning all of the field arrays from within the "storeCard" object in pass.js as empty.

## Expected behavior

headerFields, primaryFields, backFields having the actual field data from the loaded template.

## Steps to reproduce

```
      const pass = new PKPass({
        "icon.png": icon_png,
        "icon@2x.png": icon_x2_png,
        "logo.png": logo_png,
        "logo@2x.png": logo_x2_png,
        "pass.json": Buffer.from(pass_template),
        "strip@1x.png": strip_png,
      },
      {
        wwdr,
        signerCert,
        signerKey,
        signerKeyPassphrase
      }, {
        // keys to be added or overridden
        serialNumber: "AAGH44625236dddaffbda"
      });

      pass.type = "storeCard";
      
      const buffer = pass.getAsBuffer();
      fs.writeFileSync(`${model_path}/created/pass-${Date.now()}.zip`, buffer);
```

## Other details

I ran a JS debugger on my sample script, and it led me to `passkit-generator/lib/PKPass.js`.

Inside the function that starts with `[(_a = propsSymbol, _b = localizationSymbol, _c = passTypeSymbol, importMetadataSymbol)](data) {` towards the end, is an 'else' block, and I can see that it is behaving as expected here.

 ```
           const { headerFields = [], primaryFields = [], secondaryFields = [], auxiliaryFields = [], backFields = [], transitType, } = data[type] || {};
            this.headerFields.push(...headerFields);
            this.primaryFields.push(...primaryFields);
            this.secondaryFields.push(...secondaryFields);
            this.auxiliaryFields.push(...auxiliaryFields);
            this.backFields.push(...backFields);
            if (this.type === "boardingPass") {
                this.transitType = transitType;
            }
```

The expanded arrays from `data[type]` contain the data from my template.
The getters (`this.headersField`, `this.primaryFields`, etc) are working as expected, and updating the underlying class data.

When I dump `this[propsSymbol]` to the console, while still inside this function, it shows that the fields should be there.

However, once I create a stream, and I hit the line `const passJson = buffer_1.Buffer.from(JSON.stringify(this[propsSymbol]));`
The resulting buffer does not contain the fields and instead I see this being written to the pass.json file in the stream.

`"storeCard":{"headerFields":[],"primaryFields":[],"secondaryFields":[],"auxiliaryFields":[],"backFields":[]}`

Thanks in advance, and PS, I LOVE this repo !!!!

**Comment by alexandercerutti** [comment]
Hey @djpimp2010, thanks for using passkit-generator and I'm glad that you are loving it!

From a quick look, you are assigning the type. Assigning the type, as per the documentation, resets the fields. If you already have "storeCard" inside your pass.json, you don't need to set it.

That is needed if you don't provide a `pass.json`.

Let me know!

**Comment by djpimp2010** [comment]
Hi @alexandercerutti  thanks for such a fast response.

Turns out, I only added the `pass.type = "storeCard";` definition, when I was receiving an odd `value must be of type 'object' `error.
I think the cause of that was not defining `serialNumber` in my `pass.json` template or the overrides at the end of the init config.

I have now successfully signed a pass, with the fields included - so thank you for your help <3
